website/docs/guides/customize-schema-alias.md

```diff
@@ -0,0 +1,486 @@
+
+:::note Key concept
+
+The key concept is that models run from two different contexts need to result in different objects in the data warehouse to avoid people overriding each others models. For example running by developer 1 or by developer 2; or running by developer 1 and running in production.
+
+:::
+
+
+When those macros are customized, we often leverage some of the following logic:
+
+In dbt Cloud, we recommend environment variables as they can be set at the environment and all jobs will automatically inherit those values.
+
+- Set up some [environment variables](/docs/build/environment-variables) and having some Jinja (`if/else/endif`) logic to identify if the run happens in dev, prod, CI and more so.
+    
+- Or as an alternative to environment variables, you can use `target.name`. For more information, you can refer to [About target variables](/reference/dbt-jinja-functions/target). 
+
+
+<Lightbox src="/img/docs/dbt-cloud/using-dbt-cloud/Environment Variables/custom-schema-env-var.png" title="Custom schema environmental variables target name." />
+
+To allow the database/schema/object name to depend on the current branch, you can use the out of the box `DBT_CLOUD_GIT_BRANCH` environment variable in dbt Cloud [special environment variables](/docs/build/environment-variables#special-environment-variables).
+
+
+## Example use cases
+
+Here are some typical examples we've encountered with dbt users leveraging those 3 macros and different logic. 
+
+:::note
+
+Note that the following examples do not comprehensively cover all the available options. These examples are meant to be templates for you to develop your own behaviors. 
+
+:::
+
+
+
+### 1. Using the custom schema without the concatenation with the target schema in production
+
+
+The most common use case is using the custom schema without concatenating it with the default schema name when in production.
+
+To do so, you can create a new file called `generate_schema_name.sql` under your macros folder with the following code:
+
+```Jinja
+
+{% macro generate_schema_name(custom_schema_name, node) -%}
+
+    {%- set default_schema = target.schema -%}
+    {%- if custom_schema_name is none -%}
+
+        {{ default_schema }}
+
+    {%- elif  env_var('DBT_ENV_TYPE','DEV') == 'PROD' -%}
+        
+        {{ custom_schema_name | trim }}
+
+    {%- else -%}
+
+        {{ default_schema }}_{{ custom_schema_name | trim }}
+
+    {%- endif -%}
+
+{%- endmacro %}
+
+```
+
+This will generate the following outputs for a model called `my_model` with a custom schema of `marketing`, preventing any overlap of objects between dbt runs from different contexts.
+
+
+| Context     |Target database| Target schema | Resulting object               |
+|-------------|:-------------:|:-------------:|:------------------------------:|
+| Developer 1 | dev           | dbt_dev1      |dev.dbt_dev1_marketingmy_model  |
```

Comment:
I think we're missing a "." in the Resulting Object! 

`dev.dbt_dev1_marketing.my_model`
`dev.dbt_dev2_marketing.my_model`